dashboard/views.py

```diff
@@ -55,22 +54,48 @@ def gpa_map(grade):
     else:
         return GRADE_LOW
 
-def get_current_week_number(request, course_id=0):
-    # get current term start date
-    term_date_start = AcademicTerms.objects.course_date_start(course_id).date()
+def get_course_info(request, course_id=0):
+    """Returns JSON data about a course
+    
+    :param request: HTTP Request
+    :type request: Request
+    :param course_id: Unizin Course ID, defaults to 0
+    :param course_id: int, optional
+    :return: JSON to be used 
+    :rtype: str
+    """
 
-    today = datetime.now().date()
+    today = datetime.today()
 
-    logger.info(term_date_start)
-    logger.info(today)
+    try:
+        course = Course.objects.get(id=course_id)
+    except ObjectDoesNotExist:
+        return HttpResponse("{}")
 
-    ## calculate the week number
-    currentWeekNumber = math.ceil((today - term_date_start).days/7)
+    resp = model_to_dict(course)
+    # Fill in the actual term
+    term = course.term_id
 
-    # construct json
-    data = {}
-    data['currentWeekNumber'] = currentWeekNumber
-    return HttpResponse(json.dumps(data))
+    # Replace the year in the end date with start date (Hack to get around far out years)
+    # This should be replaced in the future via an API call so the terms have correct end years, or Canvas data adjusted
+    if (term.date_end.year - term.date_start.year) > 1:
+        logger.debug(f'{term.date_end.year} - {term.date_start.year} greater than 1 so setting end year to match start year.')
+        term.date_end = term.date_end.replace(year=term.date_start.year)
+
+    current_week_number = math.ceil((today - term.date_start).days/7)
+    total_weeks = math.ceil((term.date_end - term.date_start).days/7)
+    
+    resp['term'] = model_to_dict(term)
+
+    # Have a fixed maximum number of weeks
+    if total_weeks > settings.MAX_DEFAULT_WEEKS:
```

Comment:
So MAX_DEFAULT_WEEKS setting still matters. Is value of 16 good enough?

Reply:
So it looks like the total week number calculation is done whenever the file access view is generated. 

I wonder whether it is possible to cache the value, and use the same output for all courses within the same term?

Reply:
Yeah this is possible. Previously the code was calling get_current_week_number which wasn't much more efficient than this process is. (Both required a database query)

I don't believe caching is an issue specific to this, we should be utilizing [Django's built in caching. ](https://docs.djangoproject.com/en/1.11/topics/cache/)

We could easily cache to memory or filesystem. Caching to memcache would require a little more setup.

Reply:
16 is the longer number of actual weeks I've seen so far. I think we could increase this if we can ensure the start/end dates are correct, maybe a little longer to at least 20? I didn't test out the slider with more than 16 weeks though. :)

Reply:
The slider gets kind of bad out in week 20 . . . I think this might need to be redesigned to remove the "Week" name if we wanted more than 16 weeks. We might want to remove "Week" anyway as it would work better on Mobile? Maybe that's a different problem.

![image](https://user-images.githubusercontent.com/27447/50658795-7d04e900-0f68-11e9-9f49-9b8c037d266a.png)

Reply:
I created new issue about the visual here as #336

Reply:
Sounds good. I think we are done with this issue.